ChiTownTimmy23/pymadden
# 📚 : Move ruff.toml to pyproject; update justfile commands; update Ruff

## Description
This pull request makes the following changes:

### Move `ruff.toml` to `pyproject.toml`
- The `ruff.toml` file has been moved to the `pyproject.toml` file to consolidate the project's configuration files.
- This change adheres to the recommended practice of using `pyproject.toml` as the central configuration file for Python projects.
### Update Justfile Commands
- The Justfile commands have been updated to reflect the changes in the Ruff configuration file location.
- The commands now use the appropriate flags and options to specify the `pyproject.toml` file as the configuration source for Ruff.
### Update Ruff
- The version of Ruff used in the project has been updated to the latest stable release.
- This update ensures that the project is using the most recent features, bug fixes, and performance improvements provided by Ruff.

## Changes
Move `ruff.toml` to `pyproject.toml`:
- Delete the `ruff.toml` file.
- Add the Ruff configuration section to the `pyproject.toml` file.
Update Justfile commands:
- Modify the `lint` command to use the `--config pyproject.toml` flag.
- Update any other relevant commands that rely on the Ruff configuration file.
Update Ruff:
- Bump the version of Ruff in the project's dependencies to the latest stable release.
Testing
- Run the updated Justfile commands to ensure that Ruff is properly configured and functioning as expected.
- Verify that the linting process uses the configuration from the `pyproject.toml` file.
- Confirm that the updated version of Ruff is being used by checking the Ruff version output.
